### ISSUE TITLE
tf2.yml enviroment

### ISSUE THREAD

**Issue by Talkraicer** [MortenTabaka/Improving-semantic-segmentation-accuracy-for-roads-class-in-LandCover.ai-dataset#1]
Hey,
I've been trying to use conda as you have suggested in order to create the environment specified in tf2.yml
I keep getting the below error and can't find a solution:
![image](https://user-images.githubusercontent.com/81317039/185619701-ee4e6d20-5333-4e48-a683-4c19b38a3d59.png)
I'm using conda 4.12.0.
I would like some help,
Thanks,
Tal

**Comment by MortenTabaka** [comment]
Hi,
I found what's the issue and soon I will upload fixed tf2.yml or you can do quick fix in your file.

What was wrong? According to this [answer](https://stackoverflow.com/a/65964681), when I created this file via Conda, the `pip` itself was not listed for Conda to install.
Therefore the solution is adding `-pip` above `-pip:`.
```
  - pip
  - pip:
    - absl-py==0.12.0
```
I hope you will not run into other problems with the environment, since it can be sometimes moody. 

Best,
Marcin

**Comment by Talkraicer** [comment]
Hi Marcin,
sorry to bother you again.
I've seen the solution you've suggested online as well, and it did solve the warning.
But, as you can see, this isn't the main problem, but only a warning and also it says that it's adding the -pip automuatically.

The main problem is the ResolvePackageNotFound, which appears any time I run the conda env create, and it shows up with different packages each time. I googled this problem as well, and found that in order to solve it I should move the packages in the list to under the -pip: part in the tf2.yml file. I've tried this as well but still having trouble.

Do you have any suggestion to deal with that?
Thanks,
Tal

**Comment by MortenTabaka** [comment]
Hi,
I am happy to help.

The problem is caused probably by us using different operating systems. Therefore those packages builds may not be the same. I don't get `ResolvePackageNotFound` error on Ubuntu.

Moreover, there are some incosistiences in the environment

**Comment by MortenTabaka** [comment]
Hi,

The problem is probably caused by the fact that we are using different operating systems. Therefore, the compilations of these packages may not be the same. Also, there are some inconsistencies in the environment and it gives me an another error during installation.

I would recommend you to install Tensorflow from scratch. Then you should be able to run my code (maybe you will have to add a few packages).

Please take a look at this awesome tutorial from @jeffheaton, which has been a great help to me:
[Installing TensorFlow/Keras CPU/GPU w/CONDA (July, 2020)](https://www.youtube.com/watch?v=PnK1jO2kXOQ)

[Instruction on Github](https://github.com/jeffheaton/t81_558_deep_learning/blob/master/install/tensorflow-install-jul-2020.ipynb)

Although the video is from 2020, the instructions are still up-to-date and the file on Github is being consistently updated. 

Best,
Marcin

**Comment by MortenTabaka** [comment]
P.s. If you are running Tensorflow on Nvidia GPU and an error `Could not load dynamic library 'libcudnn.so.8'` would occur, then you may find solution [here](https://github.com/tensorflow/tensorflow/issues/45200#issue-751428379).

In my case command `conda install -c nvidia cudnn` solved the issue.

**Comment by MortenTabaka** [comment]
Building docker image with Docker file was introduced (tested on Ubuntu with GPU): https://github.com/MortenTabaka/Semantic-segmentation-of-LandCover.ai-dataset/tree/SSLA-5-Add-Docker-image-for-the-project#docker---tensorflow-gpu-recommended

I have in plans to implement Docker Hub - Github integration for running the project even faster.
